lxd/networks.go

```diff
@@ -258,26 +258,20 @@ func networksGet(d *Daemon, r *http.Request) response.Response {
 		return resp
 	}
 
-	allProjects := shared.IsTrue(request.QueryParam(r, "all-projects"))
-	requestProjectName := request.QueryParam(r, "project")
-
-	// requestProjectName is only valid for project specific requests.
-	if allProjects && requestProjectName != "" {
-		return response.BadRequest(errors.New("Cannot specify a project when requesting all projects"))
-	}
-
-	if requestProjectName == "" {
-		requestProjectName = api.ProjectDefaultName
-	}
-
-	// Project specific requests require an effective project, when "features.networks" is enabled this is the requested project, otherwise it is the default project.
-	effectiveProjectName, reqProject, err := project.NetworkProject(s.DB.Cluster, requestProjectName)
+	requestProjectName, allProjects, err := request.ProjectParams(r)
 	if err != nil {
 		return response.SmartError(err)
 	}
 
 	reqInfo := request.SetupContextInfo(r)
-	reqInfo.EffectiveProjectName = effectiveProjectName
+	var reqProject *api.Project
+	if !allProjects {
+		// Project specific requests require an effective project, when "features.networks" is enabled this is the requested project, otherwise it is the default project.
+		reqInfo.EffectiveProjectName, reqProject, err = project.NetworkProject(s.DB.Cluster, requestProjectName)
```

Comment:
Out of interest, should this not be done in the access control layer, I thought this was normally where EffectiveProjectName was set?